src/sources/soundsourcemodplug.cpp

```diff
@@ -0,0 +1,211 @@
+#include "sources/soundsourcemodplug.h"
+
+#include "metadata/trackmetadata.h"
+#include "util/timer.h"
+
+#include <QFile>
+
+#include <stdlib.h>
+#include <unistd.h>
+
+/* read files in 512k chunks */
+#define CHUNKSIZE (1 << 18)
+
+namespace Mixxx {
+
+namespace {
+
+QString getModPlugTypeFromUrl(QUrl url) {
+    const QString type(SoundSource::getTypeFromUrl(url));
+    if (type == "mod") {
+        return "Protracker";
+    } else if (type == "med") {
+        return "OctaMed";
+    } else if (type == "okt") {
+        return "Oktalyzer";
+    } else if (type == "s3m") {
+        return "Scream Tracker 3";
+    } else if (type == "stm") {
+        return "Scream Tracker";
+    } else if (type == "xm") {
+        return "FastTracker2";
+    } else if (type == "it") {
+        return "Impulse Tracker";
+    } else {
+        return "Module";
+    }
+}
+
+// identification of modplug module type
```

Comment:
This enum is not necessary if you set file type string based on the file name extension in `getModPlugTypeFromUrl`.